src/runtime/sched_atman.go

```diff
@@ -199,11 +199,11 @@ func (l *TaskList) Add(t *Task) {
 		atomic.StorepNoWB(unsafe.Pointer(&t.Prev), unsafe.Pointer(l.Tail))
 	} else {
 		atomic.StorepNoWB(unsafe.Pointer(&l.Head), unsafe.Pointer(t))
-		t.Prev = nil
+		// t.Prev = nil
```

Comment:
Interesting! I guess these now trigger write barriers; I tested locally and I think the correct resolution is:

```go
atomic.StorepNoWB(unsafe.Pointer(&t.Prev), nil)
```

Without clearing these, the linked list can get infinite loops.